ComponentKit/DataSources/CKCollectionViewTransactionalDataSource.mm

```diff
@@ -0,0 +1,120 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#import "CKCollectionViewTransactionalDataSource.h"
+
+#import "CKComponentInternal.h"
+#import "CKCollectionViewDataSourceCell.h"
+#import "CKTransactionalComponentDataSourceConfiguration.h"
+#import "CKTransactionalComponentDataSourceListener.h"
+#import "CKTransactionalComponentDataSourceItem.h"
+#import "CKTransactionalComponentDataSourceState.h"
+#import "CKTransactionalComponentDataSourceAppliedChanges.h"
+#import "CKComponentRootView.h"
+#import "CKComponentLayout.h"
+
+@interface CKCollectionViewTransactionalDataSource () <
+UICollectionViewDataSource,
+CKTransactionalComponentDataSourceListener
+>
+{
+  CKTransactionalComponentDataSource *_componentDataSource;
+  CKTransactionalComponentDataSourceState *_currentState;
+}
+@end
+
+@implementation CKCollectionViewTransactionalDataSource
+
+- (instancetype)initWithCollectionView:(UICollectionView *)collectionView
+                     componentProvider:(Class<CKComponentProvider>)componentProvider
+                               context:(id<NSObject>)context
+                        itemsSizeRange:(const CKSizeRange &)itemsSizeRange
+{
+  self = [super init];
+  if (self) {
+    CKTransactionalComponentDataSourceConfiguration *datasourceConfiguration = [[CKTransactionalComponentDataSourceConfiguration alloc] initWithComponentProvider:componentProvider
+                                                                                                                                                          context:context
+                                                                                                                                                        sizeRange:itemsSizeRange];
+    _componentDataSource = [[CKTransactionalComponentDataSource alloc] initWithConfiguration:datasourceConfiguration];
+    [_componentDataSource addListener:self];
+    
+    _collectionView = collectionView;
+    _collectionView.dataSource = self;
+    [_collectionView registerClass:[CKCollectionViewDataSourceCell class] forCellWithReuseIdentifier:kReuseIdentifier];
+  }
```

Comment:
Why do we need to use CKCollectionViewDataSourceCell / CKComponentRootView here, instead of being able to mount directly in the collection cell's root view?